### ISSUE TITLE
[BUG] I use quiet: true but console in chrome and in terminal not empty

### ISSUE THREAD

**Issue by MaxmaxmaximusGitHub** [zeit/next.js#12052]
# Bug report

## Describe the bug
I use `quiet: true` but console in chrome and in terminal not empty

next.config.js
```javascript
module.exports = {
  quiet: true
}
```

Steps to reproduce the behavior, please provide code snippets or a repository:

1. add `quiet: true` to next.config.js
2. run next.js project
3. console will not empty

## Expected behavior
console will be clear, cuz quiet = true, no webpack output, no info output, no events, no chunks, no "server startes in port 3000" output.

## Screenshots

![image](https://user-images.githubusercontent.com/60403187/79774403-6dffa100-833b-11ea-88ea-a7ac89fc639f.png)

![image](https://user-images.githubusercontent.com/60403187/79774419-71932800-833b-11ea-92fd-1a151eb0dad9.png)

## System information

- all systems

## Additional context

since next.js uses a lot of other libraries under the hood that write to the console both the browser and the server. Next.js should have a common unified interface for setting console output. and next.js would transparently pass these options to the libraries it uses

**Comment by mattcarlotta** [comment]
Not a bug. It appears that next uses its own custom configured webpack-dev-server. Therefore passing [devServer arguments](https://webpack.js.org/configuration/dev-server/) won't work. Instead, this feature will most likely need to be [requested](https://github.com/zeit/next.js/issues/new?template=2.Feature_request.md). That said, it appears  RFC #4808 was created several years ago to address the same issue #8422 as you've mentioned above.

**Comment by MaxmaxmaximusGitHub** [comment]
>  Therefore passing devServer arguments won't work
I do not use the arguments webpack dev server, I use the configuration next.js, therefore this is a bug, the anomalous behavior is not expected

```javascript
module.exports = {
  quiet: true
}
```

it's not a webpack option

**Comment by mattcarlotta** [comment]
Looking at an older version (v8.1.0), you're right, it looks like they are still using a custom dev-server (and not `webpack-dev-server`); however, the [quiet option (scroll down to supported options)](https://github.com/zeit/next.js/tree/0dbd3b98eca0bee01a05b8414a60c48abba76abd#custom-server-and-routing) parameter is stated as only hiding **_error_** messages derived the server (`next-dev-server`). As far as I can tell, it never silenced the [compilation messages](https://github.com/zeit/next.js/blob/0dbd3b98eca0bee01a05b8414a60c48abba76abd/packages/next/build/output/store.ts#L32-L80) on the client, but only emitted errors from the server: [1](https://github.com/zeit/next.js/blob/0dbd3b98eca0bee01a05b8414a60c48abba76abd/packages/next/server/next-dev-server.js#L151), [2](https://github.com/zeit/next.js/blob/0dbd3b98eca0bee01a05b8414a60c48abba76abd/packages/next/server/next-dev-server.js#L177). This holds true for v9.3.5 as well.

As for the HMR messages, they have never been silenced in [v8.1.0](https://github.com/zeit/next.js/blob/0dbd3b98eca0bee01a05b8414a60c48abba76abd/packages/next/client/dev-error-overlay/hot-dev-client.js#L205-L210) nor [v9.3.5](https://github.com/zeit/next.js/blob/canary/packages/next/client/dev/error-overlay/hot-dev-client.js#L240-L245), and as mentioned by Tim, they're shown for [a better development experience](https://github.com/zeit/next.js/issues/8422#issuecomment-522994547). Without them, you may not be aware of whether or not the files have been compiled/recompiled.

**Comment by mattcarlotta** [comment]
While I don't recommend using this long term, if you're looking for a quick and dirty fix, you can use the following code below to silence these messages in the browser's console (they'll still show up in a terminal):

You copy/paste this entire code into the following file (paste over the existing code)...
**node_modules/next/dist/build/output/store.js**
<details>
<summary>Click to expand minified code</summary>
<pre><code>
"use strict";exports.__esModule=true;exports.store=void 0;var _unistore=_interopRequireDefault(require("next/dist/compiled/unistore"));var _stripAnsi=_interopRequireDefault(require("next/dist/compiled/strip-ansi"));var Log=_interopRequireWildcard(require("./log"));function _getRequireWildcardCache(){if(typeof WeakMap!=="function")return null;var cache=new WeakMap();_getRequireWildcardCache=function(){return cache;};return cache;}function _interopRequireWildcard(obj){if(obj&&obj.__esModule){return obj;}if(obj===null||typeof obj!=="object"&&typeof obj!=="function"){return{default:obj};}var cache=_getRequireWildcardCache();if(cache&&cache.has(obj)){return cache.get(obj);}var newObj={};var hasPropertyDescriptor=Object.defineProperty&&Object.getOwnPropertyDescriptor;for(var key in obj){if(Object.prototype.hasOwnProperty.call(obj,key)){var desc=hasPropertyDescriptor?Object.getOwnPropertyDescriptor(obj,key):null;if(desc&&(desc.get||desc.set)){Object.defineProperty(newObj,key,desc);}else{newObj[key]=obj[key];}}}newObj.default=obj;if(cache){cache.set(obj,newObj);}return newObj;}function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}const store=(0,_unistore.default)({appUrl:null,bootstrap:true});exports.store=store;let lastStore={appUrl:null,bootstrap:true};function hasStoreChanged(nextStore){if([...new Set([...Object.keys(lastStore),...Object.keys(nextStore)])].every(key=>Object.is(lastStore[key],nextStore[key]))){return false;}lastStore=nextStore;return true;}store.subscribe(state=>{const isClient = typeof window !== undefined;if(!hasStoreChanged(state) && !isClient){return;}if(state.bootstrap){Log.wait('starting the development server ...');if(state.appUrl){Log.info(`waiting on ${state.appUrl} ...`);}return;}if(state.loading){Log.wait('compiling ...');return;}if(state.errors){Log.error(state.errors[0]);const cleanError=(0,_stripAnsi.default)(state.errors[0]);if(cleanError.indexOf('SyntaxError')>-1){const matches=cleanError.match(/\[.*\]=/);if(matches){for(const match of matches){const prop=(match.split(']').shift()||'').substr(1);console.log(`AMP bind syntax [${prop}]='' is not supported in JSX, use 'data-amp-bind-${prop}' instead. https://err.sh/zeit/next.js/amp-bind-jsx-alt`);}return;}}return;}if(state.warnings){Log.warn(state.warnings.join('\n\n'));if(state.appUrl){Log.info(`ready on ${state.appUrl}`);}return;}if(state.typeChecking){Log.info('bundled successfully, waiting for typecheck results...');return;}Log.ready('compiled successfully'+(state.appUrl?` - ready on ${state.appUrl}`:''));});
</pre></code>
</details>

All I did was add an `isClient = window !== "undefined";` variable and utilize it in an `if` statement to skip logging event messages to the browser.

To see an unminified version of the code, click [here](https://pastebin.com/fJc6PU5T) (lines 72 and 73) .

**Comment by mattcarlotta** [comment]
Taking a closer look, it appears you may be debugging from the server-side stdout in chrome's `DevTools`. When you view the DevTools attached to the `top` (`localhost:3000`) instead of `npm`, you'll only see the `console.log`s you've written within your source code (including a few HMR messages); otherwise, no server-side event (start, build, waiting, etc) messages will be displayed.

**Comment by MaxmaxmaximusGitHub** [comment]
@mattcarlotta command "next" run many node processes, webpack, server e.t.c. how i can run server instance in debug mode? Help me please =(

**Comment by mattcarlotta** [comment]
I don't use `webstorm`, so I can't help you there. Instead, I use [vscode](https://code.visualstudio.com/) and I followed this guide: https://dev.to/vvo/5-steps-to-debugging-next-js-node-js-from-vscode-or-chrome-devtools-497o#mandatory-introduction-why-even-bother.

On that note, I never run terminals through my IDE. Too difficult to manage (especially when dealing with external microservices).

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
